app/invocation/child_invocation_card.tsx

```diff
@@ -1,60 +1,57 @@
   durationMillis?: number;
 };
 
+export type InvocationMetadata =
+  | build_event_stream.WorkflowConfigured.IInvocationMetadata
+  | build_event_stream.ChildInvocationsConfigured.IInvocationMetadata;
+
 export type ChildInvocationCardProps = {
-  status: string;
-  results: BazelCommandResult[];
-  className: string;
-  icon: JSX.Element;
-  linksDisabled?: boolean;
+  result: BazelCommandResult;
 };
 
 export default class ChildInvocationCard extends React.Component<ChildInvocationCardProps> {
-  private handleCommandClicked(invocationId: string, e: React.MouseEvent) {
-    // TODO(siggisim): Switch this to using the <Link> component
-    if (e.metaKey || e.ctrlKey) {
-      return;
+  private isClickable() {
+    return this.props.result.status !== "queued" && this.props.result.status !== "not-run";
+  }
+
+  private renderStatusIcon() {
+    switch (this.props.result.status) {
+      case "succeeded":
+        return <CheckCircle className="icon" />;
+      case "failed":
+        return <XCircle className="icon" />;
+      case "in-progress":
+        return <PlayCircle className="icon" />;
+      case "queued":
+        return <Timer className="icon" />;
+      case "not-run":
+        return <CircleSlash className="icon" />;
     }
-    e.preventDefault();
-    router.navigateTo(`/invocation/${invocationId}`);
   }
 
   render() {
     return (
-      <div className={`card ${this.props.className}`}>
-        {this.props.icon}
-        <div className="content">
-          <div className="title">
-            {this.props.results.length} command{this.props.results.length === 1 || "s"} {this.props.status}
-          </div>
-          <div className="details">
-            {this.props.results.map((result) => (
-              <a
-                href={`/invocation/${result.invocation.invocationId}`}
-                className="list-grid"
-                onClick={
-                  this.props.linksDisabled
-                    ? undefined
-                    : this.handleCommandClicked.bind(this, result.invocation?.invocationId ?? "")
-                }>
-                <div className={`${!this.props.linksDisabled && "clickable"} target`}>
-                  <span className="target-status-icon">{this.props.icon}</span> {result.invocation.bazelCommand}
-                </div>
-                <div>{typeof result.durationMillis === "number" && format.durationMillis(result.durationMillis)}</div>
-              </a>
-            ))}
-          </div>
+      <Link
+        className={`child-invocation-card status-${this.props.result.status} ${this.isClickable() ? "clickable" : ""}`}
+        href={this.isClickable() ? `/invocation/${this.props.result.invocation.invocationId}` : undefined}>
+        <div className="icon-container">{this.renderStatusIcon()}</div>
+        <div className="command">{this.props.result.invocation.bazelCommand}</div>
+        <div className="duration">
+          {typeof this.props.result.durationMillis === "number" &&
```

Comment:
Can we use the question mark operator here instead of a type check? Like:
`format(duration.Millis(this.props.result.durationMillis?)`

Reply:
I think we do want to keep treating the duration as optional, so that this renders nothing for commands that haven't run yet. But I changed it to `durationMillis !== undefined` which is a little more straightforward than using `typeof durationMillis === "number"`.